Core/Core.lua

```diff
@@ -349,36 +349,39 @@ function mog:ADDON_LOADED(addon)
 		for i, model in ipairs(WardrobeCollectionFrame.ItemsCollectionFrame.Models) do
 			model:SetScript("OnMouseDown", function(self, button)
 				if IsControlKeyDown() and button == "RightButton" then
-					local link
-					local sources = WardrobeCollectionFrame_GetSortedAppearanceSources(self.visualInfo.visualID)
-					if WardrobeCollectionFrame.tooltipSourceIndex then
-						local index = WardrobeUtils_GetValidIndexForNumSources(WardrobeCollectionFrame.tooltipSourceIndex, #sources)
-						link = select(6, C_TransmogCollection.GetAppearanceSourceInfo(sources[index].sourceID))
+					local itemsCollectionFrame = self:GetParent()
+					if not itemsCollectionFrame.transmogLocation:IsIllusion() then
+						local sources = CollectionWardrobeUtil.GetSortedAppearanceSources(self.visualInfo.visualID, itemsCollectionFrame:GetActiveCategory(), itemsCollectionFrame.transmogLocation)
+						if WardrobeCollectionFrame.tooltipSourceIndex then
+							local index = WardrobeUtils_GetValidIndexForNumSources(WardrobeCollectionFrame.tooltipSourceIndex, #sources)
+							local link = select(6, C_TransmogCollection.GetAppearanceSourceInfo(sources[index].sourceID))
+							mog:AddToPreview(link)
+							return
+						end
+					else
+						mog:SetPreviewEnchant(mog:GetPreview(mog.activePreview), self.visualInfo.sourceID);
 					end
-					mog:AddToPreview(link)
-					return
 				end
 				self:OnMouseDown(button)
 			end)
 		end
-		local orig_OnMouseUp = WardrobeCollectionFrame.SetsCollectionFrame.ScrollFrame.buttons[1]:GetScript("OnMouseUp")
-		for i, button in ipairs(WardrobeCollectionFrame.SetsCollectionFrame.ScrollFrame.buttons) do
-			button:SetScript("OnMouseUp", function(self, button)
-				if IsControlKeyDown() and button == "RightButton" then
+		ScrollUtil.AddInitializedFrameCallback(WardrobeCollectionFrame.SetsCollectionFrame.ListContainer.ScrollBox, function(self, button, elementData)
+			local orig_OnClick = button:GetScript("OnClick");
+			button:SetScript("OnClick", function(self, button2)
+				if IsControlKeyDown() and button2 == "RightButton" then
 					local preview = mog:GetPreview()
 					for source in pairs(C_TransmogSets.GetSetSources(self.setID)) do
 						mog:AddToPreview(select(6, C_TransmogCollection.GetAppearanceSourceInfo(source)), preview)
 					end
 					return
 				end
-				orig_OnMouseUp(self, button)
-			end)
-		end
+				orig_OnClick(self, button2);
+			end);
+		end, self, false);
```

Comment:
I reckon we might as well pass `true` here in order to catch any previously created frames, for good measure.

Reply:
Based on my testing, nothing is actually initialized when we set up this event handler. My concern was that we'd potentially just end up rehooking all of the frames every time new ones get initialized.

Reply:
I guess we could attach a property to indicate if the frame was already initialized by mogit.

Reply:
That would happen regardless of the value this parameter if I understand correctly. But now that you mention it, the `Acquired` callback might be more suitable, unless script handlers are reset on `Initialize`.